modules/apps/map/map-common/src/main/resources/META-INF/resources/js/MapBase.js

```diff
@@ -131,7 +131,7 @@ class MapBase extends EventEmitter {
 		this._STATE_ = {
 			boundingBox,
 			controls,
-			data,
+			data: JSON.parse(data),
```

Comment:
It's curious nobody spotted this before. Initially, things make sense to me as long as data is passed as a string, in all cases.

I'm taking a broader look to make sure we don't leave out other use cases. Brb

Reply:
First of all thanks for investigating this @boton 

After a deeper look, issue is caused by [this change](https://github.com/liferay/liferay-portal/commit/01922c87ce548c38abc1ea4977933d43f3b063d5#diff-82a8c29208ff265e11df577b8829d5df98b513b4bce8dab2459bc6f402a97249L76-L78): in the process of migrating to ESM, we switched to `<liferay-frontend:component>` tag, where `points` is passed as a string and then the json syntax is escaped, making JS treat the argument as a string rather than as an object.

We have the following layers involved: 

- Before ESM migration: 
  - `mapDisplayTag` [includes] `map provider view.jsp` [invokes via aui:script] `map provider main JS class` [extends] `mapbase`
- After ESM migration: 
  - `mapDisplayTag` [includes] `map provider view.jsp` [invokes via liferay-frontend:component] `map provider main App.js` [invokes] `map provider main JS class` [extends] `mapbase`

As this is an issue caused in-between `JSP` → `App.js` communication, it seems reasonable to "fix" it as soon as data arrives to the JS (`App.js`) or even earlier (`JSP`), in any case, before things reach `mapbase`, so that code using mapbase does not need to be changed. Parsing this at `MapBase.js` level means a change in the MapBase API. This may impact on customer-provided implementations of maps that extends MapBase. I'd say this is an unlikely situation but not impossible.

I'd say [this](https://github.com/liferay/liferay-portal/blob/3d6a26e33435f578825357172a076ea8469fe802/modules/apps/map/map-openstreetmap/src/main/resources/META-INF/resources/js/App.js#L38) is a nice place to do the JSON.parse() call. 

@boton Can I add this change on top of your origin?

Reply:
@dsanz thanks for the explanations!! ❤️ 

I reviewed [these changes](https://github.com/liferay/liferay-portal/commit/01922c87ce548c38abc1ea4977933d43f3b063d5#diff-82a8c29208ff265e11df577b8829d5df98b513b4bce8dab2459bc6f402a97249L76-L78) but I discarded them because it looks safe 😕.

Of course, you can add this change but you are going to need to add in the [map google maps](https://github.com/liferay/liferay-portal/blob/7ed2a7348d92c3ebaf80b3ae94ade82e23434bc9/modules/apps/map/map-google-maps/src/main/resources/META-INF/resources/js/App.js#L38) too, my first attempt is on [map-openstreetmap/.../App.js](https://github.com/liferay/liferay-portal/blob/3d6a26e33435f578825357172a076ea8469fe802/modules/apps/map/map-openstreetmap/src/main/resources/META-INF/resources/js/App.js#L38) but I prefer not spread the JSON.parse.

Feel free to add changes on top or close it. The only problem is the time with the LPP related :(

Reply:
yep, my proposal needs change in 2 places, in compensation, the mapbase usage assumptions remain the same. It's a tradeoff in the end.